### ISSUE TITLE
Batch Inference using TensorRT T5 from demo

### ISSUE THREAD

**Issue by nrakltx** [NVIDIA/TensorRT#2459]
Hello,

I am attempting to optimize inference of T5 using TensorRT. I want to increase throughput and therefore infer using batches. My inputs don't have a constant sequence length and therefore require an attention mask to run with batches. The configured TensorRT engine in the T5 demo doesn't have attention masks as a parameter.
Is the attention mask required for the engine to infer correctly or can it be omitted like in the demo?

Thanks in advance,

N

**Comment by zerollzeng** [comment]
@nvpohanh @ttyio Can you help here?

**Comment by nvpohanh** [comment]
From our experiments, attention mask is not required for T5 inputs. You will find the batched input after tokenizer will have input_ids containing pad token (id=1), where HuggingFace’s generate() can just consume this such input_ids without attention_mask input. TensorRT follows the same behavior as HuggingFace.

**Comment by nrakltx** [comment]
Great, thank you!
